// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTBlockchainTransactionProposal";
option optimize_for = LITE_RUNTIME;

import public "BlockchainTransaction.proto";
import public "BlockchainTransactionProposedOutput.proto";
import public "BlockchainTransactionProposedNotification.proto";

message BlockchainTransactionProposal {
    optional uint32 version = 1;
    optional bytes id = 2;
    optional bytes initiator = 3;
    optional uint64 expires = 4;
    optional string memo = 5;
    repeated BlockchainTransactionProposedOutput output = 6;
    repeated BlockchainTransactionProposedNotification notification = 7;
    optional BlockchainTransaction finished = 8;
}
